### PR TITLE
[FIX] knowledge: minor fixes

### PR DESCRIPTION
**Pull request by fbarbe00**

### CODE REVIEW

File: addons/knowledge/models/knowledge_article.py [hunk 1]
```diff
@@ -583,12 +583,10 @@ def action_home_page(self):
         return action
 
     def action_set_lock(self):
-        for article in self:
-            article.is_locked = True
+        self.is_locked = True
```
Comment (julien-banken) [inline addons/knowledge/models/knowledge_article.py]: `(article_1 + article_2).action_set_lock()` :boom: ?
Maybe: `self.write({ is_locked: True })` ?

File: addons/knowledge/models/knowledge_article.py [hunk 2]
```diff
@@ -1229,9 +1228,10 @@ def _get_article_member_permissions(self, additional_fields=False):
     # MAILING
     # ------------------------------------------------------------
 
-    def _send_invite_mail(self, partner):
-        # TODO: only send one email if user is invited to multiple articles?
-        for article in self:
+    def _send_invite_mail(self, partners):
+        self.ensure_one()
+        article = self
```
Comment (dbeguin) [inline addons/knowledge/models/knowledge_article.py]: not needed. use self instead later in the code

File: addons/knowledge/models/knowledge_article_favorite.py [hunk 3]
```diff
@@ -28,50 +28,13 @@ def create(self, vals_list):
         a correct ordering as much as possible. Some sequence could be given in create values, that could lead to
         duplicated sequence per user_id. That is not an issue as they will be resequenced the next time the user reorder
         their favorites. """
-        # TODO DBE: Write/change tests to test the create multi with given sequences (that lead to duplicates) + set_sequence that should reorder eveything properly.
-        current_user_id = self.env.uid
-        favorite_without_sequences = [vals for vals in vals_list if 'sequence' not in vals]
-        user_ids = list(set(vals.get('user_id', self.env.context.get('default_user_id')) or current_user_id
-                            for vals in favorite_without_sequences))
-        max_sequence_per_user = self._get_users_max_sequence(user_ids)
+        max_sequence = self.env['knowledge.article.favorite'].search([], order='sequence desc', limit=1).sequence or 0
```
Comment (dbeguin) [inline addons/knowledge/models/knowledge_article_favorite.py]: would rename it new_sequence
and do search(...).sequence + 1 or 0
to start the sequencing at 0

File: addons/knowledge/models/knowledge_article_favorite.py [hunk 4]
```diff
@@ -28,50 +28,13 @@ def create(self, vals_list):
...
         for vals in vals_list:
-            if 'sequence' not in vals:
-                vals['sequence'] = max_sequence_per_user[vals.get('user_id', current_user_id)] + 1
-                max_sequence_per_user[vals.get('user_id', current_user_id)] += 1
-
+            if not vals.get('sequence'):
```
Comment (dbeguin) [inline addons/knowledge/models/knowledge_article_favorite.py]: here you set all the favourite to create at the same sequence (if no sequence are given)
You should increase 'new_sequence' by 1 if no sequence. If sequence is given, tant pis, there will be doublon, so be it

File: addons/knowledge/static/src/components/permission_panel/permission_panel.js [hunk 5]
```diff
@@ -87,6 +87,9 @@ class PermissionPanel extends Component {
         const index = this.state.members.findIndex(current => {
             return current.partner_id === session.partner_id;
         });
```
Comment (dbeguin) [inline addons/knowledge/static/src/components/permission_panel/permission_panel.js:89]: on peut pas faire pareil que pour l'autre ?
const index = this.state.members.indexOf(member);

File: addons/knowledge/static/src/components/permission_panel/permission_panel.js [hunk 6]
```diff
@@ -183,10 +184,8 @@ class PermissionPanel extends Component {
      * @param {Proxy} member
      */
     _onRemoveMember (event, member) {
-        const index = this.state.members.findIndex(current => {
-            return current.id === member.id;
-        });
-        if (index < 0) {
+        console.log('remove member', member);
```
Comment (dbeguin) [inline addons/knowledge/static/src/components/permission_panel/permission_panel.js]: oupsie

File: addons/knowledge/static/src/components/permission_panel/permission_panel.js (hunk 5 above)
Comment (fbarbe00) [reply addons/knowledge/static/src/components/permission_panel/permission_panel.js:89]: The issue is that in this method we don't have the member as an argument...

**Pull request closed without merging**